homeassistant/components/emoncms/__init__.py

```diff
@@ -5,22 +5,71 @@
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_API_KEY, CONF_URL, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
 
+from .const import DOMAIN, EMONCMS_UUID_DOC_URL, LOGGER
 from .coordinator import EmoncmsCoordinator
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 type EmonCMSConfigEntry = ConfigEntry[EmoncmsCoordinator]
 
 
+def _migrate_unique_id(
+    hass: HomeAssistant, entry: EmonCMSConfigEntry, emoncms_unique_id: str
+) -> None:
+    """Migrate to emoncms unique id if needed."""
+    if entry.unique_id != emoncms_unique_id:
```

Comment:
I would have kept this single line in `_check_unique_id_migration`
Then you can un-indent all this code.

Reply:
Ah yes, it makes more sense like that